### PR TITLE
Add chainlink docs

### PR DESCRIPTION
**Pull request by jpop32**
Converted and edited the Chainlink deployment tutorial by Julian Martinez .

Needs further editing, do not merge please.

### CODE REVIEW

**Comment by danlaine** [comment]
@jpop32 Just a note that you can [mark a PR as a draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/). It's a little cleaner than the "do not merge" label :)

File: build/tools/chainlink-fulfilling-requests.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+```bash
+pragma solidity ^0.6.6;
+
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/Oracle.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/MockOracle.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/Consumer.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/BasicConsumer.sol";
+```
+
+Compile the contract and you should see your repository populate
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:46]: By default Remix selects nightly Solidity build, and build fails without switching to compiler v0.6.6.

File: build/tools/chainlink-fulfilling-requests.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+```jsx
+address constant private LINK_TOKEN_POINTER = 0x0b9d5D9136855f6FEc3c0993feE6E9CE8a297846;
+```
+
+Select and compile the **Oracle.sol** file
+
+Define the LINK token contract address, 0x0b9d5D9136855f6FEc3c0993feE6E9CE8a297846
+
+and deploy!
+
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:61]: Is line 58 referring to the address we added on Line 50? Seems redundant, it might need clarification.

When I compile Oracle.sol (loaded from your gist), I get error about modifying gist. 

Deploy step should be explained better, not actually clear what needs to be done.

File: build/tools/chainlink-fulfilling-requests.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+Confirm the transaction in Metamask and record the address of your Oracle as we will need it to create a job for our node.
+
+## Creating jobs
+
+Note: Your node must be actively running. You can re-establish a connection to your active node by running the command:
+
+```bash
+ssh -i $KEY $USER@$REMOTE-IP -L 6688:localhost:6688 -N
+```
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:74]: Might want to remind users about which key/user we need here.

File: build/tools/chainlink-fulfilling-requests.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+and deploy!
+
+Now that we have our Oracle deployed, we will grant our node the ability to fulfill requests to our Oracle. By clicking the **transact** button, call the **setFulfillmentPermission** function with the address of your node, a comma, and the value true:
+
+![setFulfillmentPermission](../../.github/chainlinkp2-05-Screen_Shot_2021-07-18_at_1.16.11_PM.png)
+
+Confirm the transaction in Metamask and record the address of your Oracle as we will need it to create a job for our node.
+
+## Creating jobs
+
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:69]: Can we explain what 'Jobs' actually are and what is their purpose?

File: build/tools/chainlink-fulfilling-requests.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+}
+```
+You can also name your job. Click the **Create Job** button, and you should see a confirmation at the top of your screen!
+
+![Job confirmation](../../.github/chainlinkp2-06-Screen_Shot_2021-07-18_at_1.31.17_PM_2.png)
+
+Now our node can begin fulfilling requests!
+
+## Fulfill a request
+
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:119]: An explanation of what we're actually doing here might be helpful.

File: build/tools/chainlink-fulfilling-requests.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+Note: Changed currentPrice, price and, _price from bytes32 to Uint256
+
+View the **Consumer.sol** file and see line 21
+
+```jsx
+req.add("get", "https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD,EUR,JPY");
+```
+
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:127]: Might be a good place to explain that this is where we get the actual data and can be repurposed for other data.

File: build/tools/chainlink-fulfilling-requests.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+```
+You can also name your job. Click the **Create Job** button, and you should see a confirmation at the top of your screen!
+
+![Job confirmation](../../.github/chainlinkp2-06-Screen_Shot_2021-07-18_at_1.31.17_PM_2.png)
+
+Now our node can begin fulfilling requests!
+
+## Fulfill a request
+
+Note: Changed currentPrice, price and, _price from bytes32 to Uint256
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:120]: It is unclear why is this here and what the reader need to do with this information. Could be moved to where this is more relevant?

File: build/tools/chainlink-fulfilling-requests.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+We will use Chainlink's "**get**" request on the end point "[**https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD,EUR,JPY**](https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD,EUR,JPY)" to get ETH/USD price data.
+
+End points and data parameters can be modified using "paths". Please see [Job Specifications](https://docs.chain.link/docs/job-specifications/)  and [Fulfilling Requests](https://docs.chain.link/docs/fulfilling-requests/) for more info.
+
+We must convert our jobSpecId into Bytes32 for our contact to read it. Navigate to your created job and then click on the **Definition** tab, find your jobSpecId and convert it into bytes32 using a tool of your choice. I recommend using [https://web3-type-converter.onbrn.com/](https://web3-type-converter.onbrn.com/). Be sure to use the jobSpecID **without** dashes. In this example case, it looks like this: `95b514ab9b6449b1aa55f389ae517b5a`.
+
+![SpecID](../../.github/chainlinkp2-07-Screen_Shot_2021-07-18_at_1.40.49_PM_2.png)
+
+Select and compile the **BasicConsumer.sol** file. Enter the LINK, ORACLE, and SPECID parameters. Press `transact` to deploy!
+
```
Comment (jpop32) [inline build/tools/chainlink-fulfilling-requests.md:137]: Compilation and deployment are not entirely clear, some more guidance could be helpful.

**Review by jpop32 — COMMENTED** [review]
Added comments.

File: build/tools/chainlink-node-deployment.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+Run the Commands
+
+```bash
+sudo amazon-linux-extras install -y docker
+sudo systemctl start docker
+sudo gpasswd -a $USER docker
+exit
+```
+
+Note: The last command will log you out of the instance. When this happens, simply reconnect.
```
[the remark this answers is not in the document]
Comment (Julian-dev28) [reply build/tools/chainlink-node-deployment.md:92]: Yes you will. Then reopen the session

File: build/tools/chainlink-node-deployment.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+![DB name](../../.github/chainlinkp1-10-Screen_Shot_2021-07-15_at_7.38.24_AM.png)
+
+Reconnect to your terminal instance and edit your .env file to enter your actual database connection string in the DATABASE_URL section. Using the parameters like in the above screenshots, it would be:
+
+```bash
+DATABASE_URL=postgresql://postgres:notmyrealpassword@fuji1-db.cyovstjp94og.us-east-2.rds.amazonaws.com:5432/fuji1
+```
+
+Save the file and double-check that the environment file looks correct. Now that our Environment file is configured we can start our Chainlink node!
+
```
[the remark this answers is not in the document]
Comment (Julian-dev28) [reply build/tools/chainlink-node-deployment.md:154]: Yes we can do that

File: build/tools/chainlink-node-deployment.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+### Chainlink operator GUI
+
+Now that we have our node up and running, we can access the Chainlink node operator GUI. Open a new terminal on your local device and CD into the directory containing your Keyfile we downloaded earlier (`.pem` file). Next, Run the command `chmod 400 <your key file>` to set the correct permissions.
+
+Next, run the command:
+
+```bash
+ssh -i $KEY $USER@$REMOTE-IP -L 6688:localhost:6688 -N
+```
```
[the remark this answers is not in the document]
Comment (Julian-dev28) [reply build/tools/chainlink-node-deployment.md:185]: Ah I see. I believe I do point to how to get the information from AWS. But can change

File: build/tools/chainlink-fulfilling-requests.md (hunk 1 above)
Comment (Julian-dev28) [reply build/tools/chainlink-fulfilling-requests.md:46]: got it. we can throw a note in there

File: build/tools/chainlink-fulfilling-requests.md [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+```bash
+pragma solidity ^0.6.6;
+
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/Oracle.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/MockOracle.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/Consumer.sol";
+import "https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.6/tests/BasicConsumer.sol";
+```
+
```
Comment (Julian-dev28) [inline build/tools/chainlink-fulfilling-requests.md:45]: _Note: By default Remix selects nightly Solidity build so you must change the compiler to v0.6.6_

File: build/tools/chainlink-fulfilling-requests.md (hunk 2 above)
Comment (Julian-dev28) [reply build/tools/chainlink-fulfilling-requests.md:61]: I see. This step is redundant but it is needed to ensure the parameters are correct

I loaded the Repo from the Gist and compiled, Worked fine for me. I will try a couple more instances and see what happens

Got it. I'll expand on this point more

**Review by Julian-dev28 — COMMENTED** [review]
Please view the updated files 

Node deployment:
https://github.com/Julian-dev28/avalanche-docs/blob/add-chainlink-docs/build/tools/chainlink-node-deployment.md

Fulfilling requests:
https://github.com/Julian-dev28/avalanche-docs/blob/add-chainlink-docs/build/tools/chainlink-fulfilling-requests.md
